backend/app/controllers/spree/admin/promotions_controller.rb

```diff
@@ -41,8 +41,8 @@ def collection
 
         @collection = super
         @search = @collection.ransack(params[:q])
-        @collection = @search.result(distinct: true).
-          includes(promotion_includes).
+        @collection = @search.result.
+          includes(promotion_includes).references(:codes).
```

Comment:
I'd let this addition go into a new line, just to keep consistency with existing code